File: app/src/main/java/com/elseboot3909/GCRClient/Adapter/LabelListAdapter.java
package com.elseboot3909.GCRClient.Adapter;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.elseboot3909.GCRClient.databinding.LabelsListBinding;

public class LabelListAdapter extends RecyclerView.Adapter<LabelListAdapter.ViewHolder> {

    public static class ViewHolder extends RecyclerView.ViewHolder {

        final LabelsListBinding binding;

        public ViewHolder(LabelsListBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    public LabelListAdapter() {

    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 0;
    }



}

File: app/src/main/java/com/elseboot3909/GCRClient/Adapter/LabelListAdapter.kt
package com.elseboot3909.GCRClient.Adapter

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.elseboot3909.GCRClient.Entities.LabelPreview
import com.elseboot3909.GCRClient.R
import com.elseboot3909.GCRClient.databinding.LabelsListBinding
import java.util.*

class LabelListAdapter(private var labelsPreview: ArrayList<LabelPreview>) : RecyclerView.Adapter<LabelListAdapter.ViewHolder>() {

    inner class ViewHolder(val binding: LabelsListBinding) : RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = LabelsListBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val labelPreview = labelsPreview[position]
        holder.binding.label.text = labelPreview.label
        val total = labelPreview.votes
        when {
            total < 0 -> {
                holder.binding.total.text = "-$total"
                holder.binding.statusPic.setImageResource(R.drawable.ic_no_votes)
            }
            total > 0 -> {
                holder.binding.total.text = "+$total"
                holder.binding.statusPic.setImageResource(R.drawable.ic_vote_good)
            }
            else -> {
                holder.binding.total.text = "No votes"
                holder.binding.statusPic.setImageResource(R.drawable.ic_no_votes)
            }
        }
    }

    override fun getItemCount(): Int = labelsPreview.size

}
